docs(errors): document the different behaviour depending on status code (#1359)
api-platform/docs: core/errors.md

  "hydra:description": "The product \"1234\" does not exist."
}
```
### Message scope

> Depending on the status code you use, the message may be replaced with a generic one in production to avoid leaking unwanted informations.

Reviewer: ```suggestion
Depending on the status code you use, the message may be replaced with a generic one in production to avoid leaking unwanted information.
```

> - If your status code is >= 500 and < 600, it will be only thrown in debug environments (dev and test). In production, a message matching the status code provided will be thrown. If you are using an unofficial HTTP code, a generic message will be thrown instead.

Reviewer: ```suggestion
- If your status code is >= 500 and < 600, the exception message will only be displayed in debug mode (dev and test). In production, a generic message matching the status code provided will be shown instead. If you are using an unofficial HTTP code, a general message will be displayed.
```

  "hydra:description": "The product \"1234\" does not exist."
}
```

## Message Scope

> Depending on the status code you use, the message may be replaced with a generic one in production to avoid leaking unwanted information.
> - If your status code is >= 500 and < 600, the exception message will only be displayed in debug mode (dev and test). In production, a generic message matching the status code provided will be shown instead. If you are using an unofficial HTTP code, a general message will be displayed.
> - in any other cases, your exception message will be sent to end users

Reviewer: ```suggestion
Depending on the status code you use, the message may be replaced with a generic one in production to avoid leaking unwanted information.
If your status code is >= 500 and < 600, the exception message will only be displayed in debug mode (dev and test). In production, a generic message matching the status code provided will be shown instead. If you are using an unofficial HTTP code, a general message will be displayed.

In any other cases, your exception message will be sent to end users.
```
